harness/determined/common/api/_rbac.py

```diff
@@ -0,0 +1,136 @@
+from typing import Dict, List, Optional, Tuple
+
+from determined.common import api
+from determined.common.api import bindings
+
+
+def role_name_to_role_id(session: api.Session, role_name: str) -> int:
+    # No need to use read-paginated since the number of roles is fixed and small.
+    req = bindings.v1ListRolesRequest(limit=500, offset=0)
+    resp = bindings.post_ListRoles(session=session, body=req)
+    for r in resp.roles:
+        if r.name == role_name and r.roleId is not None:
+            return r.roleId
+    raise api.errors.BadRequestException(f"could not find role name {role_name}")
+
+
+def create_assignment_request(
```

Comment:
I don't like this method. It's got all kinds of weird. It's not your job, but do you want to be the person on the other side of the git blame?

I think it should be split into:
```python
create_user_assignment_request(session, role, workspace, username) -> bindings.v1UserRoleAssignment
create_group_assignment_request(session, role, workspace, group) -> bindings.v1GroupRoleAssignment
```

Reply:
lol, you are exactly right.  I didn't even read the function body.